domain/secret/service/service.go

```diff
@@ -368,6 +370,15 @@ func (s *SecretService) UpdateCharmSecret(ctx context.Context, uri *secrets.URI,
 	}
 	rotatePolicy := domainsecret.MarshallRotatePolicy(params.RotatePolicy)
 	p.RotatePolicy = &rotatePolicy
+	if params.RotatePolicy.WillRotate() {
+		md, err := s.GetSecret(ctx, uri)
```

Comment:
This logic was copied from the original apiserver facade, but it is inefficient - it loads the entire secret with queries using several joins just to get the rotate policy. For the new dqlite approach, there should be a new state method which returns just the rotate policy (and maybe expire time for use below).